### PR TITLE
Upgrade Storage requests to >= 0.11

### PR DESCRIPTION
**Pull request by Everlag**
This continues the migration to the new client api.

As in the PubSub upgrade, this breaks compatibility with the
previous excon based approach. All Mocks are left unimplemented.

All header comments have been rewritten to follow yard syntax.
Previously, comment headers followed a mixed style.

### CODE REVIEW

File: test/integration/storage/storage_shared.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  def setup
+    @client = Fog::Storage::Google.new
+    # Ensure any resources we create with test prefixes are removed
+    Minitest.after_run do
+      delete_test_resources
+    end
+  end
+
+  def delete_test_resources
+    unless @some_temp_file.nil?
```
Comment (Everlag) [inline test/integration/storage/storage_shared.rb]: @icco @Temikus Going to NAK this as it requires ruby 2.3.

Also, why is hound running 3 times for one PR?

Comment (icco) [inline test/integration/storage/storage_shared.rb]: Don't know why hound is running multiple times, but it runs after every push. Could you add a rule to rubocop yml to remove this output?

**Comment by icco** [comment]
Could you merge in the conflicts?

**Comment by Everlag** [comment]
Done.

File: test/integration/storage/storage_shared.rb (hunk 1 above)
Comment (Everlag) [inline test/integration/storage/storage_shared.rb]: I'll send in a send in a separate PR to master.

Comment (Everlag) [inline test/integration/storage/storage_shared.rb]: Opened in #232

File: lib/fog/storage/google_json/real.rb [hunk 2]
```diff
@@ -61,3 +61,24 @@ def signature(params)
     end
   end
 end
+
+module Fog
```
Comment (icco) [inline lib/fog/storage/google_json/real.rb]: Why are you putting this here?

File: test/integration/storage/test_buckets.rb [hunk 3]
```diff
@@ -1,55 +1,89 @@
...
 
-class TestBuckets < FogIntegrationTest
-  begin
-    @@connection = Fog::Storage::Google.new
-    @@connection.delete_bucket("fog-smoke-test")
-  rescue Exception => e
-    # puts e
+class TestStorageRequests < StorageShared
+  def test_put_bucket
+    sleep(1)
```
Comment (icco) [inline test/integration/storage/test_buckets.rb]: What's with the sleeps?

File: lib/fog/storage/google_json/real.rb (hunk 2 above)
Comment (Everlag) [inline lib/fog/storage/google_json/real.rb]: Something broke, removed.

File: test/integration/storage/test_buckets.rb (hunk 3 above)
Comment (Everlag) [inline test/integration/storage/test_buckets.rb]: Bucket operations are [strictly rate limited](https://cloud.google.com/storage/docs/best-practices#buckets-object). Is there a more idiomatic option available to rate limit tests?

Comment (icco) [inline test/integration/storage/test_buckets.rb]: Not that I know of... Usually what we do is use wait_for, and make sure the code deals with the rate limit. I'm down for that to be a future feature request tho.

**Pull request merged**